src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java

```diff
@@ -205,23 +206,35 @@ protected boolean checkBiomes() {
         }
         return false;
     }
-    
+
     /*
-		Check if player has an effect that should prevent skybox from showing
+		Check if an effect that should prevent skybox from showing
      */
     protected boolean checkEffect() {
-    	ClientPlayerEntity player = MinecraftClient.getInstance().player;
-    	Objects.requireNonNull(player);
-    	Collection<StatusEffectInstance> activeEffects = player.getStatusEffects();
-    	if (!activeEffects.isEmpty()) {
-    		for (StatusEffectInstance statusEffectInstance : Ordering.natural().reverse().sortedCopy(activeEffects)) {
-    			StatusEffect statusEffect = statusEffectInstance.getEffectType();
-    			if (statusEffect.equals(StatusEffects.BLINDNESS)) {
-    				return false;
-    			}
-    		}
-    	}
-    	return true;
+        MinecraftClient client = MinecraftClient.getInstance();
+        Objects.requireNonNull(client.world);
+        Objects.requireNonNull(client.player);
+
+        Camera camera = client.gameRenderer.getCamera();
+        boolean thickFog = client.world.getDimensionEffects().useThickFog(MathHelper.floor(camera.getPos().getX()), MathHelper.floor(camera.getPos().getY())) || client.inGameHud.getBossBarHud().shouldThickenFog();
+        if (thickFog)
+            return false;
+
+        CameraSubmersionType cameraSubmersionType = camera.getSubmersionType();
+        if (cameraSubmersionType == CameraSubmersionType.POWDER_SNOW || cameraSubmersionType == CameraSubmersionType.LAVA)
+            return false;
+
+        ClientPlayerEntity player = client.player;
+        Collection<StatusEffectInstance> activeEffects = player.getStatusEffects();
+        if (!activeEffects.isEmpty()) {
+            for (StatusEffectInstance statusEffectInstance : Ordering.natural().reverse().sortedCopy(activeEffects)) {
```

Comment:
Why do we care about the order of the effects in the instance?  Seems like a lot of work is cut by just avoiding the sort.

Reply:
Not sure why it was there previously, I can update it.